Ruby-Project/person_class.rb

```diff
@@ -23,4 +25,31 @@ def of_age?
 
     end
   end
+
+  def correct_name
+    @name
+  end
+end
+
+class Decorator < Nameable
+  def initialize(nameable)
+    super()
+    @nameable = nameable
+  end
+
+  def correct_name
+    @nameable.correct_name
+  end
+end
+
+class CapitalizeDecorator < Decorator
+  def correct_name
+    super.capitalize
+  end
+end
+
+class TrimmerDecorator < Decorator
+  def correct_name
+    super[0..9]
+  end
```

Comment:
- [ ] **[OPTIONAL]** These classes should be in their own files. Each class is usually located in its own file, according to Ruby conventions.